packages/typo3-guides-cli/src/Command/MigrateSettingsCommand.php

```diff
@@ -0,0 +1,306 @@
+    private array $unmigratedSettings = [];
+
+    /** @var int The number of successfully converted settings */
+    private int $convertedSettings = 0;
+
+    protected function configure(): void
+    {
+        $this->setDescription('Migrates Settings.cfg to guides.xml format.');
+        $this->setHelp(
+            <<<'EOT'
+                    The <info>%command.name%</info> command migrates a Settings.cfg in side the
+                    specified input directory, tries to parse it and convert all known settings
+                    to the XML format used in the guides.xml file.
+
+                    <info>$ php %command.name% [input]</info>
+
+                    EOT
+        );
+        $this->setDefinition([
+            new InputArgument(
+                'input',
+                InputArgument::REQUIRED,
+                'Path to the "Documentation" directory where Settings.cfg is stored, and guides.xml will be generated.',
+            ),
+
+            new InputOption(
+                'force',
+                null,
+                InputOption::VALUE_NONE,
+                'When set, overwrites the guides.xml file, if it exists.'
+            ),
+        ]);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $settingsFile = $input->getArgument('input') . '/Settings.cfg';
+        $guidesFile = $input->getArgument('input') . '/guides.xml';
+
+        if (!file_exists($settingsFile) || !is_readable($settingsFile)) {
+            $output->writeln('<error>Could not locate or open ' . $settingsFile . '</error>');
+            return Command::FAILURE;
+        }
+
+        if (file_exists($guidesFile) && !$input->getOption('force')) {
+            $output->writeln('<error>Target file already exists in specified directory (' . $guidesFile . ')</error>');
+            return Command::FAILURE;
+        }
+
+        $output->writeln('Migrating ' . $settingsFile . ' to ' . $guidesFile . ' ...');
+
+        if (!$this->convertSettingsToGuide($output, $settingsFile, $guidesFile)) {
+            $output->writeln('<error>Settings could not be converted. Please check for proper syntax.</error>');
+            return Command::FAILURE;
+        }
+
+        $output->writeln('Settings converted. You can now delete Settings.cfg and add guides.xml to your repository.');
+
+        return Command::SUCCESS;
+    }
+
+    /**
+     * Setup basic DOM Document
+     **/
+    private function createXmlSectionGuides(): \DOMElement
+    {
+        // Add static <guides> element with proper XMLNS
+        $guides = $this->xmlDocument->createElement('guides');
+        $guides->setAttribute('xmlns', 'https://www.phpdoc.org/guides');
+        $guides->setAttribute('xmlns:xsi', 'http://www.w3.org/2001/XMLSchema-instance');
+        $guides->setAttribute('xsi:schemaLocation', 'https://www.phpdoc.org/guides vendor/phpdocumentor/guides-cli/resources/schema/guides.xsd');
```

Comment:
```suggestion
        $guides->setAttribute('xsi:schemaLocation', 'https://www.phpdoc.org/guides ../vendor/phpdocumentor/guides-cli/resources/schema/guides.xsd');
```

This way the xsd is found.

Reply:
This will never properly resolve, because guides.xml might be in the root, or in other subdirectories.

At least for the root guides.xml this is aimed, there "../" would break it...

Reply:
Or wait, if its relative to the URI it might work and I misunderstood it. Will investigate.

Reply:
Adjusted to be relative after discussion :-)